pkg/utils/workloadidentity/secret_test.go

```diff
@@ -0,0 +1,244 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package workloadidentity
```

Comment:
The test seems to be interacting only with exposed APIs of the package, therefore you can add `_test` suffix to the package name 
```suggestion
package workloadidentity_test
```